src/cmakeBuildRunner.ts

```diff
@@ -1,172 +1,71 @@
 
 nls.config({ messageFormat: nls.MessageFormat.bundle, bundleFormat: nls.BundleFormat.standalone })();
-const localize: nls.LocalizeFunc = nls.loadMessageBundle();
 
-/**
- * Class implementing Build Runner
- *
- */
 export class CMakeBuildRunner {
 
-    private _targets?: string[];
-    private _buildPreset: preset.BuildPreset | null = null;
-    private _taskExecutor: vscode.TaskExecution | null = null;
-    private _currentBuildProcess: proc.Subprocess | null = null;
-    private _buildRunning: boolean = false;
+    private taskExecutor: vscode.TaskExecution | null = null;
+    private currentBuildProcess: proc.Subprocess | null = null;
+    private isBuildInProgress: boolean = false;
 
     constructor() {
-        this._buildRunning = false;
+        this.isBuildInProgress = false;
     }
 
-    public isBuildRunning(): boolean {
-        return this._buildRunning;
+    public buildInProgress(): boolean {
+        return this.isBuildInProgress;
     }
 
-    public setBuildRunning(running: boolean): void {
+    public setBuildInProgress(running: boolean): void {
         if (running) {
-            this._buildRunning = true;
+            this.isBuildInProgress = true;
         } else {
-            this._buildRunning = false;
-            this._targets = undefined;
-            this._buildPreset = null;
+            this.isBuildInProgress = false;
         }
     }
 
-    public async execute(buildPreset: preset.BuildPreset | null, targets?: string[]): Promise<void> {
-        const task = await this._findBuildTask(buildPreset, targets);
-
-        if (task) {
-            this._buildPreset = buildPreset;
-            this._targets = targets;
-            this._taskExecutor = await vscode.tasks.executeTask(task);
-
-            this._currentBuildProcess =  { child: undefined, result: new Promise<proc.ExecutionResult>(resolve => {
-                const disposable = vscode.tasks.onDidEndTask(e => {
-                    if (e.execution.task.group?.id === vscode.TaskGroup.Build.id) {
-                        disposable.dispose();
-                        this._taskExecutor = null;
-                        resolve({ retc: 0, stdout: '', stderr: '' });
-                    }
-                });
-            }) };
-            return;
-        }
-        this._currentBuildProcess = { child: undefined, result: new Promise<proc.ExecutionResult>((resolve) => {
-            resolve({ retc: 0, stdout: '', stderr: '' });
-        }) };
+    public async setBuildProcessForTask(taskExecutor: vscode.TaskExecution): Promise<void> {
+        this.taskExecutor = taskExecutor;
+        this.currentBuildProcess =  { child: undefined, result: new Promise<proc.ExecutionResult>(resolve => {
+            const disposable: vscode.Disposable = vscode.tasks.onDidEndTask((endEvent: vscode.TaskEndEvent) => {
+                if (endEvent.execution.task.group === vscode.TaskGroup.Build && endEvent.execution === this.taskExecutor) {
```

Comment:
From what I have managed to find you cannot compare task.group with vscode.TaskGroup.Build
You need to compare "id" fields from these two.
This is because task.group is not exactly the same object.